### ISSUE TITLE
Schema aggregator misses columns added via Schema::table() in some migrations

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#513]
## Description

The schema aggregator fails to detect columns added via `Schema::table()` (ALTER TABLE migrations) for some tables, causing false-positive `UndefinedMagicPropertyFetch` and `UndefinedMagicPropertyAssignment` errors on model properties that exist in the database.

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) with plugin v4.2.0 + Psalm 7.0.0-beta17.

### Missing model columns from `Schema::table()` migrations

**`User::$contact_sort_order`** — added in a later migration:
```php
// database/migrations/2023_05_06_125432_add_contact_sort_order_to_users.php
return new class extends Migration {
    public function up(): void {
        Schema::table('users', function (Blueprint $table) {
            $table->string('contact_sort_order')
                ->default(User::CONTACT_SORT_ORDER_LAST_UPDATED)
                ->after('name_order');
        });
    }
};
```

**`ContactTask::$distant_uuid`, `$distant_etag`, `$distant_uri`** — added in:
```php
// database/migrations/2025_08_09_223220_update_contact_important_date_and_tasks.php
Schema::table('contact_tasks', function (Blueprint $table) {
    $table->string('distant_uuid', 256)->after('vcalendar')->nullable();
    $table->string('distant_etag', 256)->after('distant_uuid')->nullable();
    $table->string('distant_uri', 2096)->after('distant_etag')->nullable();
});
```

**`Post::$slice_of_life_id`** — added in a migration that also creates another table:
```php
// database/migrations/2022_12_15_004442_create_slices_of_life_table.php
Schema::create('slice_of_lives', function (Blueprint $table) { ... });
Schema::table('posts', function (Blueprint $table) {
    $table->foreignIdFor(SliceOfLife::class)->nullable()->after('journal_id')
        ->constrained()->nullOnDelete();
});
```

**`Template::$can_be_deleted`** — added in a later migration:
```php
// database/migrations/2023_10_02_195229_add_can_be_deleted_to_templates.php
Schema::table('templates', function (Blueprint $table) {
    $table->boolean('can_be_deleted')->default(true)->after('name_translation_key');
});
```

**`SyncToken::$timestamp`** — defined in `Schema::create()` but still missed:
```php
// database/migrations/2018_12_29_135516_create_synctokens.php
Schema::create('sync_tokens', function (Blueprint $table) {
    $table->id();
    $table->foreignIdFor(User::class)->constrained()->cascadeOnDelete();
    $table->foreignIdFor(AddressBookSubscription::class, 'addressbook_subscription_id')
        ->constrained()->cascadeOnDelete();
    $table->timestamp('timestamp');
    $table->timestamps();
});
```

**`User::$two_factor_confirmed_at`** — inside a conditional:
```php
if (Fortify::confirmsTwoFactorAuthentication()) {
    $table->timestamp('two_factor_confirmed_at')
        ->after('two_factor_recovery_codes')
        ->nullable();
}
```

## Expected Behavior

All columns defined in migrations (both `Schema::create()` and `Schema::table()`) should be detected as model properties.

## Actual Behavior

These columns produce `UndefinedMagicPropertyFetch` or `UndefinedMagicPropertyAssignment` errors.

## Impact

Monica has 69 models and 74 migration files. Plugin v4.2 correctly resolves most columns (reduced from 1,889 to 62 `UndefinedMagicPropertyFetch` vs v3.1.5), but these remaining gaps affect real code paths. The `two_factor_confirmed_at` conditional pattern is especially common since it comes from the default Fortify migration.

## Notes

Some of these may share a root cause (e.g., `Schema::table()` not merging into existing table state under certain conditions). The `SyncToken::$timestamp` case is particularly interesting since it uses `Schema::create()`, not `Schema::table()`.
